lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -286,7 +306,14 @@ def unlink(self, missing_ok: bool = False):
         monkeypatch.setattr(Path, "unlink", unlink)
         with pytest.raises(Exception) as e:
             cm.create(source_tarball)
-        assert str(e.value) == "I'm broken"
+        if errno:
+            assert isinstance(e.value, OSError), f"Wanted OSError, got {type(e.value)}"
+            assert e.value.errno == errno
+        else:
+            assert isinstance(
+                e.value, Exception
+            ), f"Wanted Exception, got {type(e.value)}"
```

Comment:
I added this `assert` as an afterthought because the two branches seemed unbalanced. But you're right, it probably doesn't buy much, and I don't like `black`'s wrapping anyway...